pkg/nuctl/command/parse.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2023 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package command
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+
+	"github.com/nuclio/nuclio/pkg/nuctl/command/common"
+
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/nuclio/errors"
+	"github.com/nuclio/logger"
+	"github.com/spf13/cobra"
+)
+
+type parseCommandeer struct {
+	cmd            *cobra.Command
+	rootCommandeer *RootCommandeer
+	reportFilePath string
+	onlyFailed     bool
+	output         string
+}
+
+func newParseCommandeer(ctx context.Context, rootCommandeer *RootCommandeer) *parseCommandeer {
+	commandeer := &parseCommandeer{
+		rootCommandeer: rootCommandeer,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "parse [options]",
+		Aliases: []string{"bu"},
+		Short:   "Parse report",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// initialize root
+			if err := rootCommandeer.initialize(); err != nil {
+				return errors.Wrap(err, "Failed to initialize root")
+			}
+
+			return ParseReport(ctx, rootCommandeer.loggerInstance, commandeer.reportFilePath, commandeer.onlyFailed, commandeer.output)
+		},
+	}
+
+	cmd.Flags().StringVar(&commandeer.reportFilePath, "report-file-path", "nuctl-import-report.json", "Path to report")
+	cmd.Flags().BoolVar(&commandeer.onlyFailed, "only-failed", false, "Show only failures")
+	cmd.Flags().StringVarP(&commandeer.output, "output", "o", "", "Path to save output")
+
+	commandeer.cmd = cmd
+
+	return commandeer
+}
+
+func ParseReport(ctx context.Context, logger logger.Logger, reportPath string, onlyFailed bool, outputPath string) error {
+	reportData, err := os.ReadFile(reportPath)
+	if err != nil {
+		return errors.Wrap(err, "Failed to read report file")
+	}
+	supportedReportTypes := []common.Report{
+		&common.ProjectReports{},
+		&common.FunctionReports{},
+	}
+
+	for _, possibleType := range supportedReportTypes {
+		if err = json.Unmarshal(reportData, possibleType); err == nil {
+			t := table.NewWriter()
+			possibleType.PrintAsTable(t, onlyFailed)
+			output := t.Render()
+			if outputPath != "" {
+				if err := os.WriteFile(outputPath, []byte(output), 0644); err != nil {
+					logger.WarnWithCtx(ctx, "Failed to write output to file",
+						"path", outputPath,
+						"error", err.Error())
+				}
+			}
+			logger.Info(t.Render())
+			return nil
```

Comment:
why not stop if `err != nil`?
you continue to run `t.Render` again and return nil even if failed.

If I pass `output-path`, I don't expect it to also be printed.

Reply:
@TomerShor because if err!=nil it just means that we used the wrong type

Reply:
I was talking about the error writing to file, not the unmarshaling.

The second point still stands - 
If I pass `--output-path`, currently I will get the parsed report in the output file, and ALSO in the stdout (info log). Is this the expected behavior? or if `output-path` is passed, don't I only want the output in the given file path?

Reply:
@TomerShor why not print it and save to file at the same time?

Reply:
No real reason, just wanted to verify this is expected

Reply:
@TomerShor I think, it's fine; 
regarding error handling - the reason why I didn't add `return err` here is - in case this command will be used during upgrade flow, it will be fail the whole upgrade flow even despite the fact that it doesn't really affect anything